From: Adrien Beraud <adrien.beraud@savoirfairelinux.com>
Date: Fri, 2 Jun 2017 11:24:12 +0200
Subject: [PATCH] Handle DNS resolution error

DNS resolution error currently causes
indefinite hangup as the callback is
not called and higher-level layers
are not informed.

This commit bubbles up the error,
allowing the request to fail.
---
 source/corvusoft/restbed/detail/socket_impl.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/source/corvusoft/restbed/detail/socket_impl.cpp b/source/corvusoft/restbed/detail/socket_impl.cpp
index 313a421..665cfe7 100644
--- a/source/corvusoft/restbed/detail/socket_impl.cpp
+++ b/source/corvusoft/restbed/detail/socket_impl.cpp
@@ -145,6 +145,8 @@ namespace restbed
                         m_is_open = true;
                         callback( error );
                     } );
+                } else {
+                    callback( error );
                 }
             } );
         }
-- 
2.11.0

